server/logic/settings_logic.js

```diff
@@ -8,11 +8,27 @@ const SETTING_COLLECTION_NAME = 'settings'
 function setDefaultSettings () {
   const defaultSettings = {
     'tournamentStage': 'practice',
-    'tournamentTitle': 'World Festival Houston 2019'
+    'tournamentTitle': 'World Festival Houston 2019',
+    'practiceAmountOfMatches': 1,
+    'rankingAmountOfMatches': 3
   }
+
   db.connection().then(connection => {
-    connection.db().collection(SETTING_COLLECTION_NAME).findOne().then(response => {
-      if (!response) {
+    connection.db().collection(SETTING_COLLECTION_NAME).find({}).toArray().then(response => {
+      const promises = []
+      if (response.length > 0) {
+        response = response[0]
+        for (const setting of Object.keys(defaultSettings)) {
+          if (!response[setting]) {
+            const toSet = {}
+            toSet[setting] = defaultSettings[setting]
+            promises.push(connection.db().collection(SETTING_COLLECTION_NAME).updateOne({}, {
+              $set: toSet
+            }))
+          }
+        }
+        return Promise.all(promises)
```

Comment:
`promises`, only gets pushed one promise in the code above. Maybe it should'nt be an array?

Reply:
@idanstark42  Why only one? the push is inside the loop, for each setting that does not exists